src/main/java/xbot/common/advantage/AKitLogger.java

```diff
@@ -24,91 +39,164 @@ public AKitLogger(IPropertySupport parent) {
         this(parent.getPrefix());
     }
 
+    /**
+     * Set the log level for this particular logger instance.
+     * Log calls made after this will have that level when checking 
+     * if they should record or not.
+     * @param level new level to set
+     */
+    public void setLogLevel(LogLevel level) {
+        this.logLevel = level;
+    }
+
     public void record(String key, byte[] value) {
+        if(this.logLevel == LogLevel.DEBUG && globalLogLevel == LogLevel.INFO) {
```

Comment:
What about a function, `private boolean shouldLog()` so each of these lines can just be `if(shouldLog())` ?

Reply:
Yup good point, I was stuck on the return not being something that could be refactored away